mersenne-31/src/lib.rs

```diff
@@ -56,3 +56,32 @@ mod x86_64_avx512;
     target_feature = "avx512f"
 ))]
 pub use x86_64_avx512::*;
+
+#[cfg(not(any(
+    all(target_arch = "aarch64", target_feature = "neon"),
+    all(
+        target_arch = "x86_64",
+        target_feature = "avx2",
+        not(all(feature = "nightly-features", target_feature = "avx512f"))
+    ),
+    all(
+        feature = "nightly-features",
+        target_arch = "x86_64",
+        target_feature = "avx512f"
+    ),
+)))]
```

Comment:
AVX-512 F implies AVX2, so this is sufficient:
```suggestion
#[cfg(not(any(
    all(target_arch = "aarch64", target_feature = "neon"),
    all(
        target_arch = "x86_64",
        target_feature = "avx2",
    ),
)))]
```

Reply:
I'll also fix the corresponding thing in monty-31/lib